apps/website/src/server/db/show-and-tell.ts

```diff
@@ -488,3 +488,22 @@ export async function deletePost(id: string, authorUserId?: string) {
   ]);
   await revalidateCache(id);
 }
+
+export async function getPostsToShow() {
+  // Find the latest entry marked as seenOnStream=true
+  const latestSeenEntry = await prisma.showAndTellEntry.findFirst({
+    where: {
+      seenOnStream: true,
+    },
+    orderBy: [...postOrderBy],
+  });
+
+  // Calculate the number of entries since the latest seen entry
+  const postsToShow = await prisma.showAndTellEntry.count({
+    where: {
+      seenOnStream: false, // Only consider entries that have not been seen on stream
```

Comment:
I think this should check that the posts counted are approved (`= approvedAt > updatedAt` which is available as `whereApproved` in that file already).